hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -635,9 +660,14 @@ private AbfsRestOperation conditionalCreateOverwriteFile(final String relativePa
             .getResponseHeader(HttpHeaderConfigurations.ETAG);
 
         try {
-          // overwrite only if eTag matches with the file properties fetched befpre
-          op = client.createPath(relativePath, true, true, permission, umask,
-              isAppendBlob, eTag, tracingContext);
+          // overwrite only if eTag matches with the file properties fetched before.
+          if (getPrefixMode() == PrefixMode.BLOB) {
```

Comment:
When AbfsClient make the decision to fork to blob or DFS API based on prefix mode ?

Reply:
Have added separate API in AbfsClient for create file. So, it doesn't need fork based on prefix mode.